syntax = "proto2";

// message InnerMessage {
//   required bytes message = 1;
//   uint32 counter = 2;
//   bytes ktGossipData = 3;
//   bytes debugInfo = 99;
// }

message KeyTransparencyLoggableData {
  optional bytes ngmPublicIdentity = 1;
  optional uint32 ngmVersion = 2;
  optional uint32 ktVersion = 3;
}

message NgmPublicIdentity {
  optional bytes publicKey = 1;
}

message PublicDevicePrekey {
  required bytes prekey = 1;
  required bytes prekeySignature = 2;
  required double timestamp = 3;
}